map.go

```diff
@@ -1180,6 +1224,29 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
+	// guessing a random start key
+	if err := haveMapIterateFromNullKey(); mi.prevKey == nil && errors.Is(err, ErrNotSupported) {
```

Comment:
Can't you achieve the same by checking EFAULT and mi.prevKey == nil below?

Reply:
I moved the test logic in the error of the first NextKeyBytes() that will always failed on old kernel.
Let me know if you want/prefer to move local helper function to the MapIterator local members